app/models/user.rb

```diff
@@ -0,0 +1,19 @@
+class User < ActiveRecord::Base
+	has_many :ideas, foreign_key: :creator_id
+	has_many :comments, foreign_key: :author_id
+	has_many :votes, foreign_key: :voter_id
```

Comment:
I think trying it out as a learning exercise would be good, even if you feel it isn't a good feature for this project. In reality, if you delete a user but want to leave their comments behind, you probably have a lot of other work to do. I think Reddit leaves their comments and replaces their username with [deleted]?